File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        
        Map <Integer, Integer> numsMap = new HashMap<>();

        for (int i = 0; i < nums.length; i++) {
            int counter = target - nums[i];

            if (numsMap.containsKey(counter)){
                return new int[] {i, numsMap.get(counter)};
            }
            numsMap.put(nums[i], i);
        }
        return new int[]{};
    }
}

File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
const twoSum = function(nums, target) {
    if (nums.length === 2) return [0,1];
    let arr = [];
    for (let i = 0; i < nums.length; i++) {
        let pair = target - nums[i];
        if (arr[pair] !== undefined) return [arr[pair], i];
        else arr[nums[i]] = i;
    }
}